### PR TITLE
Fix Apiavailability chart's legend text

### LINKED ISSUE #1005: API Availability chart legends are wrong
When the API availability is no more due to response time increase, the status message contains the configured threshold time as well. Therefore each API can have a different status message. The chart's legend is then grouped by the status message, which is wrong.

The legend should have only the following three items.

1) Available
2) Server error occurred
3) High response time

### PR DESCRIPTION
**Pull request by fazlan-nazeem**
## Purpose

fixes #1005 

Also, increases feign client timeouts

### CODE REVIEW

File: components/org.wso2.analytics.apim.widgets/APIMApiAvailability/src/APIMApiAvailabilityWidget.jsx [hunk 1]
```diff
@@ -176,14 +176,28 @@ class APIMApiAvailabilityWidget extends Widget {
...
+                if (dataUnit[0].includes(legendData[0].name)) {
+                    availableCount++;
+                } else if (dataUnit[0].includes(legendData[1].name)){
+                    responseHighCount++;
+                } else if (dataUnit[0].includes(legendData[2].name)) {
+                    serverErrorCount++;
                 }
             });
+
+            let dataModified = [];
```
Comment (pahans) [inline components/org.wso2.analytics.apim.widgets/APIMApiAvailability/src/APIMApiAvailabilityWidget.jsx]: unused variable?

File: components/org.wso2.analytics.apim.widgets/APIMApiAvailability/src/APIMApiAvailabilityWidget.jsx [hunk 2]
```diff
@@ -176,14 +176,28 @@ class APIMApiAvailabilityWidget extends Widget {
...
         const { data } = message;
+        const legendData = [{name : "Available"}, {name : "Response time is high"}, {name : "Server error occurred"}];
 
         if (data) {
-            const legendData = [];
+
+            let availableCount = 0;
+            let responseHighCount = 0;
+            let serverErrorCount = 0;
             data.forEach((dataUnit) => {
```
Comment (pahans) [inline components/org.wso2.analytics.apim.widgets/APIMApiAvailability/src/APIMApiAvailabilityWidget.jsx]: seems you are creating a new array with a existing array. use `Array.map` for this.

File: components/org.wso2.analytics.apim.widgets/APIMApiAvailability/src/APIMApiAvailabilityWidget.jsx (hunk 1 above)
Comment (fazlan-nazeem) [reply components/org.wso2.analytics.apim.widgets/APIMApiAvailability/src/APIMApiAvailabilityWidget.jsx]: it's used just beneath this line.

Comment (fazlan-nazeem) [reply components/org.wso2.analytics.apim.widgets/APIMApiAvailability/src/APIMApiAvailabilityWidget.jsx]: ok. seems like i've missed assiging it to the state.

Comment (fazlan-nazeem) [reply components/org.wso2.analytics.apim.widgets/APIMApiAvailability/src/APIMApiAvailabilityWidget.jsx]: @pahans fixed. Can you check now

**Pull request merged**